### PR TITLE
Migrate Microsoft.Extensions.CommandLineUtils to System.Commandline

### PR DESCRIPTION
**Pull request by martinrrm**
## Bug

Fixes: (Don't want to close the issue)
https://github.com/NuGet/Home/issues/10039

Spec: https://github.com/NuGet/Client.Engineering/pull/1741
 
Regression? Last working version:

## Description

This is a continuation of previous work done by Fernando, this PR is going to have the full migration, more changes are needed at this time, but the current state of this branch reflects changes that Fernando started.

- https://github.com/NuGet/NuGet.Client/pull/4957
- https://github.com/NuGet/NuGet.Client/pull/4989
- https://github.com/NuGet/NuGet.Client/pull/4730
- https://github.com/NuGet/NuGet.Client/pull/4838

It's not clear why most of the work got lost, [Fernando said that the branch with changes was](https://github.com/NuGet/Home/issues/10039#issuecomment-1061197516) : [dev-dominofire-clipackage](https://devdiv.visualstudio.com/DevDiv/_git/NuGet-NuGet.Client-Trusted?path=%2F&version=GBdev-dominofire-feature-systemcommandline&_a=history), but the branch got deleted in GitHub.

Also, some of the PR's got merged into [this branch](https://devdiv.visualstudio.com/DevDiv/_git/NuGet-NuGet.Client-Trusted?path=%2F&version=GBdev-dominofire-clipackage&_a=history) so this PR is work that was reviewed before but since it's been a while, I think it's worth checking it again.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [x] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled *Pending*
  - **OR**
  - [ ] N/A

### CODE REVIEW

File: src/NuGet.Core/NuGet.CommandLine.XPlat/NuGet.CommandLine.XPlat.csproj [hunk 1]
```diff
@@ -16,18 +16,34 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.Extensions.CommandLineUtils.Sources" PrivateAssets="All" />
+	<PackageReference Include="System.CommandLine" />
```
Comment (jeffkl) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/NuGet.CommandLine.XPlat.csproj]:
```suggestion
    <PackageReference Include="System.CommandLine" />
```

File: test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatVerifyTests.cs [hunk 2]
```diff
@@ -19,14 +21,25 @@ public class XPlatVerifyTests
         [Fact]
         public void VerifyCommandArgsParsing_MissingPackagePath_Throws()
         {
+            //var logger = new TestLogger();
```
Comment (nkolev92) [inline test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatVerifyTests.cs]: delete this?

**Review by nkolev92 — COMMENTED** [review]
This is a lot to review as a single tbh. 
Probably not the most detailed review I've done, but looks pretty reasonable.

There's some redundant code that needs removed and then i'm happy to approve.

**Comment by martinrrm** [comment]
Just found out that there is a newer version of the package with breaking changes:( I'll update the PR and reopen it later.

**Pull request closed without merging**